### PR TITLE
[gdal] Update to 3.2.2

### LINKED ISSUE #15476: [ GDAL ] GDAL Update to 3.2.1
When asking a question please also include where you looked for an answer (so we can update the documentation if needed).
the version of GDAL lib is too old,please update it

### PR DESCRIPTION
**Pull request by longhuan2018**
update gdal to 3.2.2

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@longhuan2018, thanks for the PR!

Here is CI testing failures, could  you please take a look?
gdal with x64-linux and x64-osx:
```
/bin/sh: 0: Can't open ./config.rpath
configure: error: PROJ 6 symbols not found
```
[failure logs for x64-linux (2).zip](https://github.com/microsoft/vcpkg/files/5741492/failure.logs.for.x64-linux.2.zip)

pdal with x64-windows-static:
```
gdal_d.lib(nc4hdf.c.obj) : error LNK2001: unresolved external symbol H5open
gdal_d.lib(nc4memcb.c.obj) : error LNK2001: unresolved external symbol H5open
gdal_d.lib(hdf5create.c.obj) : error LNK2001: unresolved external symbol H5open
...
```
[failure logs for x64-windows-static.zip](https://github.com/microsoft/vcpkg/files/5741495/failure.logs.for.x64-windows-static.zip)

**Comment by longhuan2018** [comment]
I will test under linux before submitting

**Comment by PhoebeHui** [comment]
@longhuan2018, here latest failure in CI test with gdal:x64-osx:
```
Undefined symbols for architecture x86_64:
  "_BIO_f_ssl", referenced from:
      _ossl_connect_common in libcurl.a(openssl.c.o)
Undefined symbols for architecture x86_64:
  "_BIO_f_ssl", referenced from:
Undefined symbols for architecture x86_64:
  "_BIO_f_ssl", referenced from:
      _ossl_connect_common in libcurl.a(openssl.c.o)
      _ossl_connect_common in libcurl.a(openssl.c.o)
Undefined symbols for architecture x86_64:
  "_BIO_f_ssl", referenced from:
  "_CFArrayAppendValue", referenced from:
      _ossl_connect_common in libcurl.a(openssl.c.o)
  "_CFArrayAppendValue", referenced from:
  "_CFArrayAppendValue", referenced from:
      _sectransp_connect_common in libcurl.a(sectransp.c.o)
      _append_cert_to_array in libcurl.a(sectransp.c.o)
      _sectransp_connect_common in libcurl.a(sectransp.c.o)
      _append_cert_to_array in libcurl.a(sectransp.c.o)
```

[failure logs for x64-osx (4).zip](https://github.com/microsoft/vcpkg/files/5746026/failure.logs.for.x64-osx.4.zip)

**Comment by JackBoosY** [comment]
Stuck on osx? I will test this PR manually.

**Comment by JackBoosY** [comment]
I can confirm that gdal is stuck in the configuration process in linux, this is the error log content:
```
  1 configure: WARNING: unrecognized options: --disable-silent-rules
  2 rm: conftest.dSYM: is a directory
  3 rm: conftest.dSYM: is a directory
  4 rm: conftest.dSYM: is a directory
  5 rm: testftrapv.dSYM: is a directory
  6 /bin/sh: ./config.rpath: No such file or directory
  7 ././configure: line 24210: 0: command not found
  8 ././configure: line 24313: 0: command not found
```

**Comment by JackBoosY** [comment]
Pinging @longhuan2018 for response. Is work still being done for this PR?

**Comment by longhuan2018** [comment]
> Pinging @longhuan2018 for response. Is work still being done for this PR?

I've reproduced the problem in OSX, but I haven't found the reason yet. The Spring Festival is coming soon, and the PR will not continue for the time being

**Comment by JackBoosY** [comment]
Please use command `./vcpkg format-manifest --all` then commit changes about `vcpkg.json`.

**Comment by longhuan2018** [comment]
@JackBoosY Finally, there is no error. Do you have time to review it?

File: ports/gdal/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "gdal",
+  "version": "3.2.1",
```
Comment (JackBoosY) [inline ports/gdal/vcpkg.json:3]:
```suggestion
  "version-semver": "3.2.1",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by PhoebeHui** [comment]
All features test passed with x64-windows.

**Review by PhoebeHui — APPROVED** [review]

**Comment by PhoebeHui** [comment]
@longhuan2018, could you resolve the conflicts?

**Comment by JackBoosY** [comment]
depends on #17277.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please run command `vcpkg x-add-version --all` then commit the changes.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
@strega-nil Since this PR lasts long time, please review and merge this PR first.

**Pull request merged**